subsys/bluetooth/host/audio/ascs.c

```diff
@@ -926,6 +1240,90 @@ static ssize_t ascs_qos(struct bt_ascs *ascs, struct net_buf_simple *buf)
 	return buf->size;
 }
 
+static bool ascs_codec_metadata_store(struct bt_data *data, void *user_data)
+{
+	struct bt_codec *codec = user_data;
+	struct bt_codec_data *meta;
+
+	if (codec->meta_count >= ARRAY_SIZE(codec->meta)) {
+		BT_ERR("No slot available for Codec Config Metadata");
+		return false;
+	}
+
+	meta = &codec->meta[codec->meta_count];
+
+	if (data->data_len > sizeof(meta->value)) {
+		BT_ERR("Not enough space for Codec Config Metadata: %u > %zu",
+		       data->data_len, sizeof(meta->value));
+		return false;
+	}
+
+	BT_DBG("#%u type 0x%02x len %u", codec->meta_count, data->type,
+	       data->data_len);
+
+	meta->data.type = data->type;
+	meta->data.data_len = data->data_len;
+
+	/* Deep copy data contents */
+	meta->data.data = meta->value;
```

Comment:
Please explain

Reply:
Can you elaborate?

This comment seemingly just refer to use doing the `memcpy` of the value, instead of just doing a pointer. 

Why it has been done this way I don't know (copied code from Luiz's implementation).

Reply:
This seems to be a copying from "data" to "meta" (see the memcpy below).
But here, there is a copy (an assignment) from a "meta" member to another "meta" member.
That is the part I do not follow.
But if this is a copy of existing code, it can be argued that it is not really a part of this PR.

Reply:
> But here, there is a copy (an assignment) from a "meta" member to another "meta" member.
That is the part I do not follow.

It is the `meta->data.data = meta->value;` you are referring to, right?
`meta->data.data` is a a pointer in a `struct bt_data`, and `meta->value` is the buffer holding the data. 
I suppose the `struct bt_data` is being used as we can then use the `bt_data_parse` function to parse the LTV structures.